zerver/lib/email_mirror.py

```diff
@@ -436,27 +436,33 @@ def process_stream_message(to: str, message: EmailMessage) -> None:
     # that all messages must have a topic.
     subject = subject or _("Email with no subject")
 
-    stream, options = decode_stream_email_address(to)
+    channel_email_address, options = decode_stream_email_address(to)
+
     # Don't remove quotations if message is forwarded, unless otherwise specified:
     if "include_quotes" not in options:
         options["include_quotes"] = is_forwarded(subject_header)
 
-    user_profile = get_system_bot(settings.EMAIL_GATEWAY_BOT, stream.realm_id)
+    channel = channel_email_address.channel
+    sender = user_profile = channel_email_address.sender
+    creator = channel_email_address.creator
+    realm = channel_email_address.realm
+    email_gateway_bot = get_system_bot(settings.EMAIL_GATEWAY_BOT, realm.id)
+
+    if sender.id == email_gateway_bot.id and creator is not None:
+        user_profile = creator
```

Comment:
This would be better done as having the `else` block set `user_profile`, and also s/user_profile/user_for_permissions/`

Reply:
> This would be better done as having the else block set user_profile

Done

> and also s/user_profile/user_for_permissions/`

I think, you meant to rename `user_profile` to something better ? -- renamed it to `user_for_access_check`.

Reply:
yes